tests/apiextensions_test.go

```diff
@@ -0,0 +1,1082 @@
+// limitations under the License.
+
+package crossplane
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	kextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/v2/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/v2/pkg/fieldpath"
+	"github.com/crossplane/crossplane-runtime/v2/pkg/resource"
+	"github.com/crossplane/crossplane-runtime/v2/pkg/resource/unstructured/composite"
+
+	extv1 "github.com/crossplane/crossplane/v2/apis/apiextensions/v1"
+	extv2 "github.com/crossplane/crossplane/v2/apis/apiextensions/v2"
+	pkgv1 "github.com/crossplane/crossplane/v2/apis/pkg/v1"
+
+	"github.com/crossplane/conformance/internal"
+)
+
+// TestCompositeResourceDefinitionNamespace tests the creation of a namespaced XRD and
+// verifies that the resulting CRD is conformant.
+func TestCompositeResourceDefinitionNamespace(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	t.Cleanup(cancel)
+
+	kube, err := internal.NewClient()
+	if err != nil {
+		t.Fatalf("Create client: %v", err)
+	}
+
+	// create a v2 XRD
+	xrd := &extv2.CompositeResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{Name: "namespaceconformances.test.crossplane.io"},
+		Spec: extv2.CompositeResourceDefinitionSpec{
+			Scope: extv2.CompositeResourceScopeNamespaced,
+			Group: "test.crossplane.io",
+			Names: kextv1.CustomResourceDefinitionNames{
+				Kind:     "NamespaceConformance",
+				ListKind: "NamespaceConformanceList",
+				Plural:   "namespaceconformances",
+				Singular: "namespaceconformance",
+			},
+			Versions: []extv2.CompositeResourceDefinitionVersion{{
+				Name:          "v1alpha1",
+				Served:        true,
+				Referenceable: true,
+				Schema: &extv2.CompositeResourceValidation{
+					OpenAPIV3Schema: runtime.RawExtension{Raw: []byte("{}")},
+				},
+			}},
+		},
+	}
+
+	if err := kube.Create(ctx, xrd); err != nil {
+		t.Fatalf("Create XRD %q: %v", xrd.GetName(), err)
+	}
+	t.Logf("Created XRD %q", xrd.GetName())
+
+	t.Cleanup(func() {
+		t.Logf("Cleaning up XRD %q.", xrd.GetName())
+		if err := kube.Get(ctx, types.NamespacedName{Name: xrd.GetName()}, xrd); err != nil {
+			t.Fatalf("Get XRD %q: %v", xrd.GetName(), err)
+		}
+		if err := kube.Delete(ctx, xrd); err != nil {
+			t.Fatalf("Delete XRD %q: %v", xrd.GetName(), err)
+		}
+		t.Logf("Deleted XRD %q", xrd.GetName())
+	})
+
+	// verify the XRD becomes established
+	testXRDIsEstablished(ctx, t, kube, xrd)
+
+	t.Run("CRDIsCreatedForXR", func(t *testing.T) {
```

Comment:
I wonder if we should test that you _can't_ do legacy stuff (e.g. use claims) when using v2 style XRs. e.g. Test hat only `LegacyCluster` XRs make claim CRDs.

Reply:
good idea! i'll include a test case for this as well, to ensure that modern XRDs do not produce claim CRDs.